tests/unit/test_configuration.py

```diff
@@ -304,19 +334,23 @@ def test_overriding_nonexistent_keys_fails(config, match):
         get_configuration(config)
 
 
-def get_noise_type_configs(noise_names: list):
-    configs = list(itertools.product(noise_names, PROBABILITY_VALUE_LOGS))
+def get_noise_type_configs(noise_names: list, noise_type: str):
+    configs = list(itertools.product(noise_names, PROBABILITY_VALUE_LOGS[noise_type]))
     return [(x[0], x[1][0], x[1][1]) for x in configs]
 
 
 @pytest.mark.parametrize(
     "row_noise_type, value, match",
-    get_noise_type_configs(ROW_NOISE_TYPES),
+    get_noise_type_configs(ROW_NOISE_TYPES, "row_noise"),
 )
 def test_validate_standard_parameters_failures_row_noise(row_noise_type, value, match):
     """
     Tests valid configuration values for probability for row noise types.
     """
+    key = False
+    if row_noise_type.name == NOISE_TYPES.duplicate_with_guardian.name:
+        key = True
```

Comment:
What is `key` here?

Reply:
key is the key for the value logs dict.  So value_logs[False] = match and then for guardian duplication we make the key True.